### PR TITLE
reverse cashOutTax logic

### PR DESCRIPTION
**Pull request by mejango**
# Description

*What does this PR: do, how, why?*

## Limitations & risks

*Are there any trade-off or new vulnarbility surface based on theses changes?*

# Check-list
- [ ] Tests are covering the new feature
- [ ] Code is [natspec'd](https://docs.soliditylang.org/en/v0.8.17/natspec-format.html)
- [ ] Code is [linted and formatted](https://docs.soliditylang.org/en/v0.8.17/style-guide.html)
- [ ] I have run the test locally (and they pass)
- [ ] I have rebased to the latest main commit (and tests still pass)

# Interactions
These changes will impact the following contracts:
- Directly:

- Indirectly:

### CODE REVIEW

File: src/JBMultiTerminal.sol [hunk 1]
```diff
@@ -1045,6 +1045,134 @@ contract JBMultiTerminal is JBPermissioned, ERC2771Context, IJBMultiTerminal {
...
+                memo: ""
+            });
+        }
+
+        // Keep a reference to the amount being reclaimed that is subject to fees.
+        uint256 amountEligibleForFees;
+
+        // Send the reclaimed funds to the beneficiary.
+        if (reclaimAmount != 0) {
+            // Determine if a fee should be taken. Fees are not exercised if the cash out tax rate is at its max (100%),
```
Comment (xBA5ED) [inline src/JBMultiTerminal.sol:1116]: Comment is no longer correct

File: src/libraries/JBCashOuts.sol [hunk 2]
```diff
@@ -25,7 +25,7 @@ library JBCashOuts {
         returns (uint256)
     {
         // If the cash out tax rate is 0, no surplus can be reclaimed.
```
Comment (xBA5ED) [inline src/libraries/JBCashOuts.sol:27]: Comment is no longer correct

**Review by xBA5ED — COMMENTED** [review]
There are some comments that still reflect the old situation

**Comment by mejango** [comment]
> There are some comments that still reflect the old situation

squashed em

**Review by xBA5ED — APPROVED** [review]

**Pull request merged**